apps/backoffice/src/app/api/institutions/[institutionId]/groups/route.ts

```diff
@@ -0,0 +1,53 @@
+import { handleInternalErrorResponse } from "@/lib/be/errors";
+import { retrieveUserGroups } from "@/lib/be/institutions/business";
+import { sanitizedNextResponseJson } from "@/lib/be/sanitize";
+import { withJWTAuthHandler } from "@/lib/be/wrappers";
+import { parseStringToNumberFunction } from "@/utils/string-util";
+import { NextRequest, NextResponse } from "next/server";
+
+import { BackOfficeUser } from "../../../../../../types/next-auth";
+
+/**
+ * @description Retrieve groups for an Institution ID
+ */
+export const GET = withJWTAuthHandler(
+  async (
+    request: NextRequest,
+    {
+      backofficeUser,
+      params
+    }: { backofficeUser: BackOfficeUser; params: { institutionId: string } }
+  ) => {
+    if (backofficeUser.institution.role !== "ADMIN") {
```

Comment:
Will this check be done also in future controllers? if so do you think is bettere moving it in a dedicated module in order to be reusable?

Reply:
[6feebeb](https://github.com/pagopa/io-services-cms/pull/1007/commits/6feebeb6a8dde1db5f3d5ed287c21699585aee06)